source/sync/event-library.txt

```diff
@@ -0,0 +1,319 @@
+such as when following a link or looking an object up by primary key.
+
+The Event Library records read events as a JSON object with two fields:
+
+- ``type``: stores the class name
+- ``value``: stores an array of serialized objects
+
+Read events store values as follows:
+
+- *Single-object read events*: the value is an array that has a single element
+- *Objects matching a query*: the value is an array of all objects matching
+  a query, even if the objects are never used
+- *Reads that occur during a write transaction*: the value is the data that
+  the object has before the write transaction began; it does not reflect
+  any changes that occur during the write event.
+- *Objects that do not exist when a write transaction begins*: objects that
+  are created in a write transaction do not produce a read event at all.
+
+.. important::
+
+   The Event Library cannot tell if only a subset of the query displays in 
+   the client application. For example, say the client application has a 
+   list view. The Event Library's read event doesn't record scrolling 
+   information; it records the read event as the full query result. 
+   Developers must use custom events to record when a client application 
+   displays only a subset of a query result.
+
+Read Event Combining
+````````````````````
+
+A stream of every read event could produce a lot of "duplicate" events on
+the same objects that don't add information. To reduce these "duplicate" 
+events, the Event Library discards and merges some events.
+
+The Event Library discards:
+
+- Queries which match no objects
+- Queries which match only newly-created objects
+- Object reads where the object is matched by a previous query
+
+The Event Library merges:
+
+- Multiple queries on the same table a merged into a single query.
+
+Embedded Object Read Events
+```````````````````````````
+
+The Event Library represents embedded objects by creating a link in the 
+parent object to the primary key of the embedded object. When the user
+does not follow the link, the primary key is the only representation of
+the embedded object. When the user does follow the link, the embedded 
+object resolves within the parent object.
+
+This also produces a top-level object read.
+
+.. example::
+
+   .. code-block:: json
+      :caption: Parent Object with an Unfollowed Link
+   
+      {
+        "type": "Person",
+        "value": [{
+          "_id": "1-2-3",
+          "name": "Frodo Baggins",
+          "address": "4-5-6"
+        }]
+      }
+
+   .. code-block:: json
+      :caption: Parent Object with a Followed Link
+   
+      {
+        "type": "Person",
+        "value": [{
+          "_id": "1-2-3",
+          "name": "Frodo Baggins",
+          "address": [{
+            "_id": "4-5-6",
+            "street": "1 Bag End",
+            "city": "Hobbiton Hill",
+            "state": "The Shire"
+          }]
+        }]
+      }
+
+Read Event Format
```

Comment:
Should we move this above the other Read Event subheadings? This seems like the most relevant and the others use this shape too.